docs/pages/reference/join-methods.mdx

```diff
@@ -0,0 +1,427 @@
+will require the node to join again to renew its certificate.
+
+The join method and its parameters are specified in the Token.
+
+### Token
+
+A Token is a Teleport resource that specifies which join method can be used in
+which context. For example, a token can allow SSH services to join with the
+`iam` join method if they are in the AWS account `333333333333` and can assume
+the role `teleport-node-role`:
+
+```yaml
+kind: token
+version: v2
+metadata:
+  name: my-iam-token
+spec:
+  roles: [Node]
+  join_method: iam
+  allow:
+  - aws_account: "333333333333"
+    aws_arn: "arn:aws:sts::333333333333:assumed-role/teleport-node-role/i-*"
+```
+
+<Admonition type="important">
+The token name may, or may not be sensitive depending on the join method.
+Secret-based join methods rely on the token name to be secret. In such
+cases the token name must be protected as knowing the token name in enough
+for a node to join the cluster.
+</Admonition>
+
+## Classification of join methods
+
+```mermaid
+mindmap
+  root((Join Methods))
+    secret(Secret-based <br/> and Renewable Certs)
+      Static Tokens in the config
+      ::icon(fa fa-triangle-exclamation)
+      Ephemeral Tokens
+    delegated(Delegated)
+      rejoin{{Non-renewable Certs}}
+        Cloud Provider
+            iam
+            azure
+            gcp
+        CI Provider
+            circleci
+            github
+            jenkins
+            gitlab
+        Kubernetes
+      oneshot{{Renewable Certs}}
+        ec2
+```
+
+### Static vs Dynamic
+
+A static join method is a method that relies on a long-lived hardcoded token.
+Static secret tokens are supported by Teleport for backward compatibility reasons
+and their usage is strongly discouraged.
+
+All the other join methods are dynamic: their tokens are created though the API
+and can either be long-lived and delegated, or secret and short-lived.
+Dynamically-created tokens can sometimes be called Provision Tokens.
+
+### Secret vs Delegated
+
+#### Secret-based join methods
+
+Secret-based join methods are universal. The joining node sends the secret and the
+Auth validates that it matches the one it knows. Those joining methods are
+inherently prone to secret exfiltration and the Delegated join methods should
+be preferred when available. If you have to use a secret-based join method, it is
+recommended to use short-lived tokens (valid only 1 hour for example) to reduce
+the risk of the token leaking.
+
+Secret-based join methods are:
+- static `token`
+- ephemeral `token`
+
+#### Delegated Join methods
+
+Delegated join methods rely on the context of the joining node and a third party to
+establish trust. The third party can be a cloud provider, a CI platform or the
+container runtime. Those methods cannot be used for every node (e.g. joining an SSH
+agent from a Raspberry Pi is not possible) but should be preferred when possible.
+
+Delegated join methods might also offer more granularity. For example, cloud-provider
```

Comment:
Yes, and each method is more or less granular depending on the 3rd party it relies on. For example, Azure joining is a mess because they put very little information in the identity document, while AWS or GHA lets you write more granular conditions.